scanpy/external/tl/_trimap.py

```diff
@@ -0,0 +1,148 @@
+from ... import logging as logg
+import numpy as np
+import scipy.sparse as scp
+
+
+def trimap(
+    adata,
+    n_dims=2,
+    n_inliers=10,
+    n_outliers=5,
+    n_random=5,
+    metric='euclidean',
+    weight_adj=500.0,
+    lr=1000.0,
+    n_iters=400,
+    init_pos=None,
+    verbose=False,
+    copy=False,
+):
+    """\
+    TriMap: Large-scale Dimensionality Reduction Using Triplets.
+
+    TriMap is a dimensionality reduction method that uses triplet constraints
+    to form a low-dimensional embedding of a set of points. The triplet
+    constraints are of the form "point i is closer to point j than point k".
+    The triplets are sampled from the high-dimensional representation of the
+    points and a weighting scheme is used to reflect the importance of each
+    triplet.
+
+    TriMap provides a significantly better global view of the data than the
+    other dimensionality reduction methods such t-SNE, LargeVis, and UMAP.
+    The global structure includes relative distances of the clusters, multiple
+    scales in the data, and the existence of possible outliers. We define a
+    global score to quantify the quality of an embedding in reflecting the
+    global structure of the data.
+
+    Reference
+    ---------
+
+    @article{2019TRIMAP,
+     author = {{Amid}, E. and {Warmuth}, M. K.},
+     title = "{TriMap: Large-scale Dimensionality Reduction Using Triplets}",
+     journal = {ArXiv e-prints},
+     archivePrefix = "arXiv",
+     eprint = {1910.00204},
+     year = 2019,
+    }
+
+    Input
+    ------
```

Comment:
Please call this section `Parameters`, use numpydoc style and get rid of “default” (it’s added automatically)